service/src/main/java/bio/terra/drshub/services/DrsApiFactory.java

```diff
@@ -45,36 +44,20 @@ public DrsApi getApiFromUriComponents(UriComponents uriComponents, DrsProvider d
     return new DrsApi(drsClient);
   }
 
-  RestTemplate makeRestTemplateWithPooling() {
-    PoolingHttpClientConnectionManager poolingConnManager =
-        new PoolingHttpClientConnectionManager();
-    poolingConnManager.setMaxTotal(CONNECTION_POOL_SIZE);
-    poolingConnManager.setDefaultMaxPerRoute(CONNECTION_POOL_SIZE);
-    CloseableHttpClient httpClient =
-        HttpClients.custom().setConnectionManager(poolingConnManager).build();
-    HttpComponentsClientHttpRequestFactory factory =
-        new HttpComponentsClientHttpRequestFactory(httpClient);
-    return new RestTemplate(factory);
-  }
-
-  public RestTemplate makeMTlsRestTemplateWithPooling(String certPath, String keyPath) {
-    var keyManager = PemUtils.loadIdentityMaterial(certPath, keyPath);
-    var sslFactory = SSLFactory.builder().withIdentityMaterial(keyManager).build();
-    var socketFactory = Apache4SslUtils.toSocketFactory(sslFactory);
-    Registry<ConnectionSocketFactory> socketFactoryRegistry =
-        RegistryBuilder.<ConnectionSocketFactory>create().register("https", socketFactory).build();
-
-    PoolingHttpClientConnectionManager poolingConnManager =
-        new PoolingHttpClientConnectionManager(socketFactoryRegistry);
-    poolingConnManager.setMaxTotal(CONNECTION_POOL_SIZE);
-    poolingConnManager.setDefaultMaxPerRoute(CONNECTION_POOL_SIZE);
-    CloseableHttpClient httpClient =
-        HttpClients.custom()
-            .setSSLSocketFactory(socketFactory)
-            .setConnectionManager(poolingConnManager)
-            .build();
-    HttpComponentsClientHttpRequestFactory factory =
-        new HttpComponentsClientHttpRequestFactory(httpClient);
-    return new RestTemplate(factory);
+  private RestTemplate getOrCreateRestTemplate(DrsProvider drsProvider) {
+    var name = drsProvider.getName();
+    if (restTemplateCache.containsKey(name)) {
```

Comment:
It'd be better to avoid work here if possible. Normally `.containsKey()` is lightweight but in a synchronized map it requires a mutex lock.
```suggestion
    if (log.isDebugEnabled() && restTemplateCache.containsKey(name)) {
```